src/main/java/com/google/jenkins/plugins/computeengine/ComputeEngineRetentionStrategy.java

```diff
@@ -67,6 +67,8 @@ public void start(ComputeEngineComputer c) {
   @Override
   public void taskAccepted(Executor executor, Queue.Task task) {
     if (oneShot) {
+      // When a oneshot instance is used only one task is run, so better not accept more.
+      ((ComputeEngineComputer) executor.getOwner()).setAcceptingTasks(false);
```

Comment:
Cool..

Reply:
I'm concerned about the potential for race conditions here. Perhaps an atomic get/set is called for. Something like this might help address:
`
synchronized(this) {
  ComputeEngineComputer node = (ComputeEngineComputer) executor.getOwner();
  if (!node.isAcceptingTask()) {
  // fail gracefully
  }
  node.setAcceptingTasks(false);
}
`